server/src/test/java/com/adobe/testing/s3mock/its/ListObjectIT.java

```diff
@@ -91,13 +95,11 @@ static Param param(String prefix, String delimiter) {
         param(null, "").keys(ALL_OBJECTS), //
         param("/", null), //
         param("b", null).keys("b", "b/1", "b/1/1", "b/1/2", "b/2"), //
-        param("b/", null).keys("b/1", "b/1/1", "b/1/2", "b/2"), //
+        param("b/", null).keys("b/1/1", "b/1/2"), //
         param("b", "/").keys("b").prefixes("b/"), //
-        param("b/", "/").keys("b/1", "b/2").prefixes("b/1/"), //

```

Comment:
I don't understand why you would want to remove this and the two cases below. They test valid situations.

Reply:
I think I figured out what was confusing me:
- the mock was doing `getParent` when `prefix` contained `File.separator`
- however, `/` has no special meaning in the prefix unless it's equal to the `delimiter`. So I ran into the issue with non-standard delimiters:

```
listV1[12: prefix=d, delimiter=:](com.adobe.testing.s3mock.its.ListObjectIT)  Time elapsed: 0.222 s  <<< FAILURE!
java.lang.AssertionError:
Returned keys are correct
Expected: an empty collection
     but: <[d:1, d:1:1]>
```

and I spent too much time trying to fix that - however, I think that should be a separate PR/issue. So I put it all back